scooby/report.py

```diff
@@ -296,6 +299,27 @@ def cols(html, version, name, ncol, i):
         return html
 
 
+class Inspection(Report):
+    """A class to inspect the active environment and generate a report based
+    on all imported modules. Simply pass the ``globals()`` dictionary.
+    """
+    def __init__(self, global_vars, additional=None, ncol=3, text_width=80,
+                 sort=False,):
+
+        stdlib_pkgs = get_standard_lib_modules()
+        modules = set()
+        for var in copy(global_vars).values():
```

Comment:
I'm wondering now if this `copy` will copy big data structures that might be in memory... e.g. if you just did some complex numerical analysis and have GBs worth of numpy arrays in memory, this copy might double the memory consumption.

Reply:
The following (ridiculous) comprehension would avoid this copying issue:

```py
        stdlib_pkgs = get_standard_lib_modules()
        modules = set()
        modules_dirty = set([val if inspect.ismodule(val) else (sys.modules[val.__module__] if (hasattr(val, "__module__") and val.__module__ != "__main__") else None) for val in global_vars.values()])
        modules_dirty.remove(None)
        for module in modules_dirty:
            name = module.__name__.split(".")[0]
            if name not in stdlib_pkgs:
                modules.add(name)
```